Sources/App/Core/PackageContributorService.swift

```diff
@@ -0,0 +1,133 @@
+// Copyright 2020-2022 Dave Verwer, Sven A. Schmidt, and other contributors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+import ShellOut
+import Vapor
+
+
+public struct Contributor {
+    /// Total number of commits
+    public let numberOfCommits: Int
+    public let email: String
+    public let name: String
+    
+}
+
+
+/// Loads the contributors history from a Git repository
+struct GitHistoryLoader {
+    
+    static func loadContributorsHistory(cacheDirPath: String, packageID: UUID?) throws -> [Contributor] {
+        do {
+            let commitHistory = try queryVCHistory(cacheDirPath: cacheDirPath, packageID: packageID)
+            return try parseVCHistory(log: commitHistory)
+        } catch {
+            throw AppError.analysisError(packageID, "loadContributorsHistory failed: \(error.localizedDescription)")
+        }
+    }
+    
+    /// Gets the version control history in a string log
+    private static func queryVCHistory(cacheDirPath: String, packageID: UUID?) throws -> String {
+        
+        if !Current.fileManager.fileExists(atPath: cacheDirPath) {
+            throw AppError.unexistentPackageCacheDir(packageID, cacheDirPath)
+        }
+
+        // attempt to shortlog
+        do {
+            return try Current.git.shortlog(cacheDirPath)
+        } catch {
+            throw AppError.shellCommandFailed("gitShortlog",
+                                              cacheDirPath,
+                                              "queryVCHistory failed: \(error.localizedDescription)")
+        }
+    }
+    
+    /// Parses the result of queryVCHistory into a collection of contributors
+    private static func parseVCHistory(log: String) throws -> [Contributor] {
+        var committers = [Contributor]()
+        
+        for line in log.components(separatedBy: .newlines) {
+            let log = line.split(whereSeparator: { $0 == " " || $0 == "\t"})
+            
+            if (log.count != 2) {
```

Comment:
What's special about two here? 

My understanding is there is at least 3 pieces of information we expect, so perhaps a >= would be better?

(Especially as we have some force unwraps in here which I suppose could crash if log.count was 0 or 1)

Reply:
Oh yes, `>2` would be better. Thank you! Under the assumption that the logs are of the form `numberOfCommits identifier email` the logic here is
1. take the first token as the number of commits
2. take the last token as the email
3. the identifier is the join of the tokens in between (some people just write one name, others full name)